### PR TITLE
configurable eth_call accepted errors

### PR DESCRIPTION
**Pull request by AlfredoG87**
**Description**:
make accepted error codes from mirror node for eth_call configurable and added initial default ones

**Related issue(s)**:

Fixes #

**Notes for reviewer**:
Whenever the mirror node returns an acceptedErrorCode  the  mirrorNode  request  will return null, and in response  we will  return empty hex.

image

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: packages/relay/tests/lib/eth.spec.ts [hunk 1]
```diff
@@ -3390,17 +3390,17 @@ describe('Eth calls using MirrorNode', async function () {
         }
       });
 
-      sdkClientStub.submitContractCallQueryWithRetry.returns({
+      /*sdkClientStub.submitContractCallQueryWithRetry.returns({
```
Comment (Nana-EC) [inline packages/relay/tests/lib/eth.spec.ts:3393]: Did you mean to remove these comments?

File: packages/server/tests/acceptance/htsPrecompile/precompileCalls.spec.ts [hunk 2]
```diff
@@ -738,12 +738,14 @@ describe('@precompile-calls Tests for eth_call with HTS', async function () {
...
 
-            await relay.callFailing(
+            /*await relay.callFailing(
                 'eth_call',
                 [callData, 'latest'],
                 predefined.CONTRACT_REVERT(),
                 requestId
-            );
+            );*/
+            const res = await relay.call('eth_call', [callData, 'latest'], requestId);
```
Comment (ebadiere) [inline packages/server/tests/acceptance/htsPrecompile/precompileCalls.spec.ts:747]: So these are meant to be enabled again I assume, correct?

**Review by ebadiere — COMMENTED** [review]
One question around the commented out test relay calls.

File: packages/relay/tests/lib/eth.spec.ts (hunk 1 above)
Comment (Nana-EC) [reply packages/relay/tests/lib/eth.spec.ts:3393]: Also when you don't expect a fallback to consensus not would be good to ensure this is not called

File: packages/relay/src/lib/clients/mirrorNodeClient.ts [hunk 3]
```diff
@@ -297,6 +304,9 @@ export class MirrorNodeClient {
         const acceptedErrorResponses = MirrorNodeClient.acceptedErrorStatusesResponsePerRequestPathMap.get(pathLabel);
         if (error.response && acceptedErrorResponses && acceptedErrorResponses.indexOf(effectiveStatusCode) !== -1) {
             this.logger.debug(`${requestIdPrefix} [${method}] ${path} ${effectiveStatusCode} status`);
+            if(pathLabel  === MirrorNodeClient.CONTRACT_CALL_ENDPOINT) {
```
Comment (Nana-EC) [inline packages/relay/src/lib/clients/mirrorNodeClient.ts:307]: I think you should add a new set of tests for `handleError()`
This can confirm for `contract/call` that the codes we allow return null and the ones we don't throw the error

**Review by Nana-EC — CHANGES_REQUESTED** [review]
Add some simple `handleErrror()` tests and remove the comments.

File: packages/server/tests/acceptance/htsPrecompile/precompileCalls.spec.ts (hunk 2 above)
Comment (AlfredoG87) [reply packages/server/tests/acceptance/htsPrecompile/precompileCalls.spec.ts:747]: I will remove  the  comments.

File: packages/relay/src/lib/clients/mirrorNodeClient.ts (hunk 3 above)
Comment (Nana-EC) [reply packages/relay/src/lib/clients/mirrorNodeClient.ts:307]: Maybe
```ts
describe('handleError', async() => {

    const CONTRACT_CALL_ENDPOINT = 'contracts/call';
    const nullResponseCodes = [400,404,415,429,500];
    const errorRepsonseCodes = [501, 503];

    for (const code of nullResponseCodes) {
      it(`returns null when ${code} is returned`, async () => {
        const result = await mirrorNodeInstance.handleError(new Error('test error'), CONTRACT_CALL_ENDPOINT, code, 'POST');
        expect(result).to.equal(null);
      });
    }

    for (const code of errorRepsonseCodes) {
      it(`throws an error when ${code} is returned`, async () => {
        try {
          await mirrorNodeInstance.handleError(new Error('test error'), CONTRACT_CALL_ENDPOINT, code, 'POST');
          expect.fail('should have thrown an error');
        } catch (e) {
          expect(e.message).to.equal('test error');
        }
      });
    }
  });
```

File: packages/relay/tests/lib/eth.spec.ts (hunk 1 above)
Comment (AlfredoG87) [reply packages/relay/tests/lib/eth.spec.ts]: ✅  done

File: packages/relay/src/lib/clients/mirrorNodeClient.ts (hunk 3 above)
Comment (AlfredoG87) [reply packages/relay/src/lib/clients/mirrorNodeClient.ts:307]: Thank for  the  starting point,  good suggestion!  👍 

✅   done

File: packages/relay/tests/lib/mirrorNodeClient.spec.ts [hunk 4]
```diff
@@ -48,13 +50,44 @@ describe('MirrorNodeClient', async function () {
       },
       timeout: 20 * 1000
     });
+    //  process.env.ETH_CALL_ACCEPTED_ERRORS =
```
Comment (Nana-EC) [inline packages/relay/tests/lib/mirrorNodeClient.spec.ts]: nit: missed but leave unless you have to push another commit for something else

File: packages/server/tests/acceptance/rpc_batch3.spec.ts [hunk 5]
```diff
@@ -348,7 +348,9 @@ describe('@api-batch-3 RPC Server Acceptance Tests', function () {
                             data: '0x3ec4de3800000000000000000000000067d8d32e9bf1a9968a5ff53b87d777aa8ebbee69'
                         };
 
-                        await relay.callFailing('eth_call', [callData, 'latest'], predefined.CONTRACT_REVERT(), requestId);
+                        // await relay.callFailing('eth_call', [callData, 'latest'], predefined.CONTRACT_REVERT(), requestId);
```
Comment (Nana-EC) [inline packages/server/tests/acceptance/rpc_batch3.spec.ts]: nit: leftover

File: packages/server/tests/acceptance/htsPrecompile/precompileCalls.spec.ts (hunk 2 above)
Comment (AlfredoG87) [reply packages/server/tests/acceptance/htsPrecompile/precompileCalls.spec.ts:741]: ✅   comments  removed

**Review by Nana-EC — APPROVED** [review]
left over comments, but looks good

File: packages/server/tests/acceptance/rpc_batch3.spec.ts (hunk 5 above)
Comment (Nana-EC) [reply packages/server/tests/acceptance/rpc_batch3.spec.ts]: Seems to be removed
